telcell/celldb/pgdatabase.py

```diff
@@ -0,0 +1,414 @@
+import csv
+import datetime
+import math
+import warnings
+from typing import Optional, List, Iterable, Callable, Tuple
+
+import geopy
+
+from telcell.cell_identity import (
+    Radio,
+    CellIdentity,
+    CellGlobalIdentity,
+    EutranCellGlobalIdentity,
+)
+from . import duplicate_policy
+from .cell_collection import CellCollection, Properties
+from ..data.models import RD_TO_WGS84, WGS84_TO_RD
+from ..geography import Angle
+
+RD_X_RANGE = (7000, 300000)
+RD_Y_RANGE = (289000, 629000)
+
+
+def rd_to_point(x: int, y: int) -> geopy.Point:
+    if not (RD_Y_RANGE[0] <= y <= RD_Y_RANGE[1]) or not (
+        RD_X_RANGE[0] <= x <= RD_X_RANGE[1]
+    ):
+        warnings.warn(
+            f"rijksdriehoek coordinates {x}, {y} outside range: x={RD_X_RANGE}, y={RD_Y_RANGE}"
+        )
+    c = RD_TO_WGS84.transform(x, y)
+    return geopy.Point(longitude=c[0], latitude=c[1])
+
+
+def point_to_rd(point: geopy.Point) -> Tuple[int, int]:
+    return WGS84_TO_RD.transform(point.longitude, point.latitude)
+
+
+def _build_antenna(row):
+    date_start, date_end, radio, mcc, mnc, lac, ci, eci, rdx, rdy, azimuth_degrees = row
+    if radio == Radio.GSM.value or radio == Radio.UMTS.value:
+        retrieved_ci = CellIdentity.create(
+            radio=radio, mcc=mcc, mnc=mnc, lac=lac, ci=ci
+        )
+    elif radio == Radio.LTE.value or radio == Radio.NR.value:
+        retrieved_ci = CellIdentity.create(radio=radio, mcc=mcc, mnc=mnc, eci=eci)
+    elif radio is not None:
+        raise ValueError(f"unrecognized radio type: {radio}")
+    elif lac is not None and ci <= 0xFFFF:
+        retrieved_ci = CellIdentity.create(
+            radio=radio, mcc=mcc, mnc=mnc, lac=lac, ci=ci
+        )
+    else:
+        retrieved_ci = CellIdentity.create(radio=radio, mcc=mcc, mnc=mnc, eci=eci)
+
+    coords = rd_to_point(rdx, rdy)
+    azimuth = Angle(degrees=azimuth_degrees) if azimuth_degrees is not None else None
+    return Properties(wgs84=coords, azimuth=azimuth, cell=retrieved_ci)
+
+
+def _build_cell_identity_query(ci):
+    qwhere = []
+    qargs = []
+    if ci.radio is not None:
+        qwhere.append("radio = %s")
+        qargs.append(ci.radio)
+    if ci.mcc is not None:
+        qwhere.append(f"mcc = {ci.mcc}")
+    if ci.mnc is not None:
+        qwhere.append(f"mnc = {ci.mnc}")
+
+    if isinstance(ci, CellGlobalIdentity):
+        if ci.lac is not None:
+            qwhere.append(f"lac = {ci.lac}")
+        if ci.ci is not None:
+            qwhere.append(f"ci = {ci.ci}")
+
+    elif isinstance(ci, EutranCellGlobalIdentity):
+        if ci.eci is not None:
+            qwhere.append(f"eci = {ci.eci}")
+
+    else:
+        raise ValueError(f"unsupported cell type: {type(ci)}")
+
+    if len(qwhere) == 0:
```

Comment:
waarom doe je dit?

Reply:
Omdat een lege `WHERE` in sql niet toegestaan is